crates/chia-consensus/src/spendbundle_conditions.rs

```diff
@@ -75,7 +77,20 @@ pub fn run_spendbundle(
         ret.execution_cost += clvm_cost;
         subtract_cost(a, &mut cost_left, clvm_cost)?;
 
-        let buf = tree_hash(a, puz);
+        cache.visit_tree(a, puz);
+        let buf = if flags & COST_SHATREE != 0 {
+            tree_hash_cached_costed(
+                a,
+                puz,
+                &mut cache,
+                &mut cost_left,
+                constants.shatree_recurse_cost,
+                constants.shatree_cost_per_byte,
```

Comment:
I'm not sure we necessarily need to add these to `ConsensusConstants`. I don't think we put any other cost-related constants in there. Except cost_per_byte I suppose. But condition cost are not in there, nor any other clvm execution costs